src/test/java/io/dropwizard/grpc/server/GrpcServerTests.java

```diff
@@ -0,0 +1,242 @@
+
+        runTestWithConfig(invalidPortYamlConfig);
+    }
+
+    @Test
+    public void createsPlainTextServer() throws Exception {
+        final DropwizardTestSupport<TestConfiguration> testSupport =
+                new DropwizardTestSupport<>(TestApplication.class, resourceFilePath("grpc-test-config.yaml"));
+
+        ManagedChannel channel = null;
+        try {
+            testSupport.before();
+            channel = createPlaintextChannel(testSupport);
+            final PersonServiceGrpc.PersonServiceBlockingStub client = PersonServiceGrpc.newBlockingStub(channel);
+
+            final GetPersonResponse resp =
+                    client.getPerson(GetPersonRequest.newBuilder().setName(TEST_PERSON_NAME).build());
+            assertEquals(TEST_PERSON_NAME, resp.getPerson().getName());
+        } finally {
+            testSupport.after();
+            shutdownChannel(channel);
+        }
+    }
+
+    @Test
+    public void createsServerWithTls() throws Exception {
+        final DropwizardTestSupport<TestConfiguration> testSupport = new DropwizardTestSupport<>(TestApplication.class,
+            resourceFilePath("grpc-test-config.yaml"), Optional.empty(),
+            ConfigOverride.config("grpcServer.certChainFile", getURIForResource("cert/server.crt")),
+            ConfigOverride.config("grpcServer.privateKeyFile", getURIForResource("cert/server.key")));
+
+        ManagedChannel channel = null;
+        try {
+            testSupport.before();
+            channel = createClientChannelForEncryptedServer(testSupport);
+            final PersonServiceGrpc.PersonServiceBlockingStub client = PersonServiceGrpc.newBlockingStub(channel);
+
+            final GetPersonResponse resp =
+                    client.getPerson(GetPersonRequest.newBuilder().setName(TEST_PERSON_NAME).build());
+            assertEquals(TEST_PERSON_NAME, resp.getPerson().getName());
+        } finally {
+            testSupport.after();
+            shutdownChannel(channel);
+        }
+    }
+
+    @Test
+    public void shouldCompleteRequestIfShutdownPeriodLongerThanLongRunningRequest() throws Exception {
+        final long shutdownPeriodInMillis = 3000L;
+        final long slowRequestDurationInMillis = 1000L;
+
+        testShutdownPeriodHonoured(shutdownPeriodInMillis, slowRequestDurationInMillis,
+            (requestDurationInMillisFut, shutdownDurationInMillisFut) -> {
+                try {
+                    final long requestDurationInMillis = requestDurationInMillisFut
+                        .get(slowRequestDurationInMillis + DELTA_IN_MILLIS, TimeUnit.SECONDS);
+                    assertEquals((double) requestDurationInMillis, slowRequestDurationInMillis, DELTA_IN_MILLIS);
+                } catch (final Exception e) {
+                    throw new RuntimeException(e);
+                }
+            });
+    }
+
+    @Test
+    public void shouldNotCompleteRequestIfShutdownPeriodShorterThanLongRunningRequest() throws Exception {
```

Comment:
This testcase checks if forceful shutdown works correctly and does not seem to assert anything about the request.

Suggestion: stopsServerForcefullyWhenShutdownPeriodShorterThanLongRunningRequest

Reply:
Strictly speaking this method does not test anything about the forceful shutdown, only that the stop lifecycle event returns after the configured shutdownPeriod.